### PR TITLE
[WIP]OHARA-595 Metadata of RowConnector

### PR DESCRIPTION
**Pull request by g1geordie**
here has "a lot of mess" .  please view and give some suggestion

what I want to do  is
The Connector will bind our supportype data from front end
and then pass the object to task

We use this binding object to extract metadata and give front end info

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -129,7 +129,7 @@ subprojects {
       tests.extendsFrom archives
     }
     task testJar(type: Jar, dependsOn: testClasses) {
-      archiveClassifier.set('tests')
+//      archiveClassifier.set('tests')
```
Comment (chia7712) [inline build.gradle]: Why? Are you using gradle 4.x ? If so, please upgrade to 5.x

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/jar/JarStore.scala [hunk 2]
```diff
@@ -72,6 +72,8 @@ trait JarStore extends Releasable {
 
   def jarInfos(ids: String): Future[Seq[JarInfo]] = jarInfos().map(_.filter(p => ids.contains(p.id)))
 
+  def jarInfos(ids: Seq[String]): Future[Seq[JarInfo]] = jarInfos().map(_.filter(p => ids.contains(p.id)))
```
Comment (chia7712) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/jar/JarStore.scala]: Nice one. Seems the variant (jarInfos(ids: String)) is a wrong method. Please remove it

File: build.gradle (hunk 1 above)
Comment (g1geordie) [reply build.gradle]: Yes ,you r right . I have upgrade it

File: ohara-kafka/src/main/java/com/island/ohara/kafka/connector/TaskConfig2.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import com.island.ohara.common.annotations.Optional;
+import com.island.ohara.common.data.Column;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/** this class carries all required configs for row connectors. */
+public class TaskConfig2<T> {
```
Comment (chia7712) [inline ohara-kafka/src/main/java/com/island/ohara/kafka/connector/TaskConfig2.java]: this name is not acceptable...

Comment (chia7712) [inline ohara-kafka/src/main/java/com/island/ohara/kafka/connector/TaskConfig2.java]: Please describe the usage of this class...

**Comment by chia7712** [comment]
We need more discussion. see https://github.com/oharastream/ohara/issues/297

**Pull request closed without merging**